controller/controller.mustache

```diff
@@ -35,18 +41,28 @@ module.exports = {
     });
   },
   watch: (params) => {
+    params = controller.normaliseParams(params);
+    const queryParams = schema.filter('{{name}}', params);
+
+    const changes = {includeInitial: true, includeStates: true};
     const table = r.table('{{pluralName}}');
+    const indexBy = controller.getIndex(modelSchema, queryParams);
 
-    if (params.id) {
-      return table.get(params.id).changes({includeInitial: true, includeStates: true}).run();
+    var query = rethinkdb.buildQuery(table, indexBy, queryParams);
+    query = rethinkdb.addTransformations(query, params);
+ 
+    //if query params were passed but none were on an index field
+    if (!_.isEmpty(queryParams) && indexBy == null) {
+      console.log(NOT_INDEXED);
```

Comment:
As discussed going to leave this as is as was Dave's recommendation to warn the user via console.log. I also think it'd be good to have a better way to do this, but not sure what that would look like. We could emit a warning via sockets or something, but we'd then have to do something with it :/